C++/C++.sublime-syntax

```diff
@@ -26,2340 +28,616 @@ first_line_match: |-
-          set:
-            - meta_content_scope: meta.function.parameters.c++ meta.group.c++
-            - match : \)
-              scope: punctuation.section.group.end.c++
-              set: function-definition-continue
-            - include: modifiers
-            - include: types
-            - match: '{{identifier}}(?=\s*(\[|,|\)|=))'
-              scope: variable.parameter.c++
-            - match: '='
-              scope: keyword.operator.assignment.c++
-              push:
-                - match: '(?=,|\))'
-                  pop: true
-                - include: expressions-minus-generic-type
-                - include: scope:source.c#preprocessor-line-continuation
-            - include: expressions-minus-generic-type
-            - include: scope:source.c#preprocessor-line-continuation
-    - match: (?=\S)
-      pop: true
+        1: variable.parameter.type.c++
+      pop: 1
+    - include: expression
 
-  function-definition-continue:
-    - meta_content_scope: meta.function.c++
-    - include: comments
-    - match: '(?=;)'
-      pop: true
-    - match: '->'
-      scope: punctuation.separator.c++
-      set: function-definition-trailing-return
-    - include: function-specifiers
-    - include: constraint-requires
-    - match: '='
-      scope: keyword.operator.assignment.c++
-    - match: '&'
-      scope: keyword.operator.c++
-    - match: \b0\b
-      scope: meta.number.integer.decimal.c++ constant.numeric.value.c++
-    - match: \b(default|delete)\b
-      scope: storage.modifier.c++
-    - match: '(?=\{)'
-      set: function-definition-body
-    - match: '(?=\S)'
-      pop: true
-
-  function-definition-trailing-return:
-    - include: comments
-    - match: '(?=;)'
-      pop: true
-    - match: '(?=\{)'
-      set: function-definition-body
-    - include: function-specifiers
-    - include: constraint-requires
-    - include: function-trailing-return-type
+  type.concept:
+    - meta_scope: meta.concept.c++
+    - match: '{{identifier}}'
+      scope: entity.name.concept.c++
+      push: type.concept-body
+    - include: else-pop
+  type.concept-body:
+    - match: (?=;|\)|\]|\})
+      pop: 1
+    - include: expression
+
+  enum-def:
+    - match: \b(enum)(?:\s*(class))?\b
+      captures:
+        1: keyword.declaration.enum.c++
+        2: keyword.declaration.class.c++
+      branch_point: enum
+      branch: [type.enum, pop-immediately]
+
+  enclosure:
+    - meta_prepend: true
+    - include: lambda-enclosure
+  enclosure.brackets:
+    - meta_prepend: true
+    - match: (\])\s*(<)
+      fail: lambda
+    - match: (\])\s*(\()
+      captures:
+        1: punctuation.section.brackets.end.c++
+        2: meta.group.c++ punctuation.section.group.begin.c++
+      set: [lambda-try-not, enclosure.parenthesis]
 
-  function-definition-body:
-    - meta_content_scope: meta.function.c++ meta.block.c++
-    - match: '\{'
-      scope: punctuation.section.block.begin.c++
+    - match: (\])\s*(\{|\-\>|mutable\b)
```

Comment:
`fail`-patterns don't need capture groups nor do they need to consume text as it is rewinded anyway. Thus lookahead patterns are optimal choise.l